mcs/nunit24/NUnitCore/core/BabysitterSupport.cs

```diff
@@ -0,0 +1,104 @@
+// Added to NUnit by Andi McClure to add special support for the test babysitter script.
+// See scripts/babysitter in Mono repository.
+
+namespace NUnit.Core
+{
+	using System;
+	using System.IO;
+	using System.Collections.Generic;
+	using NUnit.Core.Filters;
+
+	public class BabysitterSupport
+	{
+		enum OverrideMode {None, Run, Exclude};
+		public static Dictionary<string, bool> OverrideTests = new Dictionary<string, bool>();
+		private static OverrideMode Override = OverrideMode.None;
+		private static string CurrentTestFile = null, RanTestFile = null, FailedTestFile = null;
+
+		private static void DeleteFile(string path)
+		{
+			try {
+				File.Delete(path);
+			} catch (Exception e) {}
+		}
+		private static void WriteFile(string path, string contents)
+		{
+			DeleteFile(path);
+			File.AppendAllText(path, contents);
+		}
+
+		// Environment variables are available from process start, so safe to do setup in a static constructor
+		static BabysitterSupport()
+		{
+			string overrideModeString = Environment.GetEnvironmentVariable("NUNIT_BABYSITTER_RUN_MODE");
+			string overrideTestString = Environment.GetEnvironmentVariable("NUNIT_BABYSITTER_RUN_TEST");
+			if (overrideModeString == "RUN")
+				Override = OverrideMode.Run;
+			else if (overrideModeString == "EXCLUDE")
+				Override = OverrideMode.Exclude;
+			if (Override != OverrideMode.None)
+			{
+				char [] semicolon = {';'};
+				foreach (string s in overrideTestString.Split(semicolon, StringSplitOptions.RemoveEmptyEntries))
```

Comment:
Without a variable the next line becomes 
`foreach (string s in overrideTestString.Split(new char[] {";"}, StringSplitOptions.RemoveEmptyEntries))` ... seems a bit overstuffed to me, would you really say that's more readable?